anchore_engine/services/catalog/catalog_impl.py

```diff
@@ -2196,32 +2191,85 @@ def _delete_image_for_real(userId, image_record, dbsession, image_ids, image_ful
         "manifest_data",
         "parent_manifest_data",
     ]:
-        archiveId = imageDigest
-        logger.debug("DELETEing image from archive {}/{}".format(bucket, archiveId))
-        rc = obj_store.delete(userId, bucket, archiveId)
+        try:
+            logger.debug("DELETEing image from archive %s/%s", bucket, image_digest)
+            rc = obj_store.delete(account_id, bucket, image_digest)
+        except Exception:
```

Comment:
the usage of try/except here could be better. It is useful to capture only what is expected to be failing - even if it is being logged.

In all instances where this try/except/log pattern is used, it is actually not needed: the exception is still caught by the caller (and again, a broad exception). Ideally a granular exception - or no exception at all would be caught. It is up to the caller to decide how to deal with this.

Introducing these patterns makes it hard to debug (can't catch these exceptions anywhere) and impossible to test accurately, because again, all exceptions are caught

Reply:
The called function - obj_store.delete(), does not transform errors into a specific type of exception, it just raises any errors caught. Without changing this function, it is impossible to handle a specific type of exception in the calling function. That's a bigger change well outside the scope of this ticket

Reply:
I see what you mean for the handling of exceptions in this particular function - whatever was the last exception will get raised later. However, it is unclear what exception we are talking about here. It certainly can't be any possible exception?